### ISSUE TITLE
gqlgen init error in loading gqlparser.

### ISSUE THREAD

**Issue by KellyLSB** [99designs/gqlgen#1402]
### What happened?
```
kellylsbkr@penguin:~/go/src/github.com/KellyLSB/angelrin$ go run github.com/99designs/gqlgen init
validation failed: packages.Load: graph/generated/generated.go:16:2: cannot find package "github.com/vektah/gqlparser/v2" in any of:
        /usr/local/go/src/github.com/vektah/gqlparser/v2 (from $GOROOT)
        /home/kellylsbkr/go/src/github.com/vektah/gqlparser/v2 (from $GOPATH)
graph/generated/generated.go:17:2: cannot find package "github.com/vektah/gqlparser/v2/ast" in any of:
        /usr/local/go/src/github.com/vektah/gqlparser/v2/ast (from $GOROOT)
        /home/kellylsbkr/go/src/github.com/vektah/gqlparser/v2/ast (from $GOPATH)
/home/kellylsbkr/go/src/github.com/KellyLSB/angelrin/graph/generated/generated.go:16:12: could not import github.com/vektah/gqlparser/v2 (invalid package name: "")
/home/kellylsbkr/go/src/github.com/KellyLSB/angelrin/graph/generated/generated.go:17:2: could not import github.com/vektah/gqlparser/v2/ast (invalid package name: "")

Exec "go run ./server.go" to start GraphQL server
```

### What did you expect?
A v2 directory within gqlparser; or... something

### Minimal graphql.schema and models to reproduce
This is the project init

### versions
 - `gqlgen version`?  master
 - `go version`? go1.15.5 linux/amd64
 - dep or go modules? neither, yet

This has not been a problem in my older application of gqlgen; merely the new init.

**Comment by diffuse** [comment]
This is referenced in #1105, if you look at the [go.mod](https://github.com/vektah/gqlparser/blob/master/go.mod) file for gqlgen, you can see that the module is defined as `github.com/vektah/gqlparser/v2`.

Like #1105 says, if you're not running in go module mode, I believe `$GOPATH/src` is searched for `github.com/vektah/gqlparser/v2`, which won't exist, since go get downloads it to `github.com/vektah/gqlparser`

If you're ok with using go module mode, then an easy fix for it is to:
- run `go mod init` in your project root, which will create a `go.mod` file 
- re-run `go run github.com/99designs/gqlgen init` which should now work without that error
- this command should also add the appropriate modules to the `require` list in the project `go.mod` file

If your project is called `foo`, your `go.mod` file will end up looking something like this:

```
module github.com/<your username>/foo

go 1.15

require (
	github.com/99designs/gqlgen v0.13.0
	github.com/vektah/gqlparser/v2 v2.1.0
)
```
When you run `go install ./...`, your other dependencies should also be automatically added to this file.

Your previous application probably used a sub-2.0 version of gqlgen, which would have the module defined without the `v2` ([go.mod](https://github.com/vektah/gqlparser/blob/v1.3.1/go.mod))

**Comment by frederikhors** [comment]
If @diffuse's answer solves it please close this issue, @KellyLSB.

**Comment by KellyLSB** [comment]
Presently:
```
kellylsbkr@penguin:~/go/src/github.com/KellyLSB/demondin$ go run github.com/99designs/gqlgen init
package github.com/99designs/gqlgen: no Go files in /home/kellylsbkr/go/src/github.com/KellyLSB/demondin/vendor/github.com/99designs/gqlgen
```

**Comment by rokiyama** [comment]
I have had the same issue, I have run `go get github.com/vektah/gqlparser/v2` but I still get the another error.

```bash
$ go run github.com/99designs/gqlgen init
validation failed: packages.Load: /Users/rokiyama/projects/gqlgen-todos/graph/prelude.resolvers.go:19:44: __DirectiveResolver not exported by package generated
```

full logs:

```bash
$ mkdir gqlgen-todos
$ cd gqlgen-todos
$ go mod init github.com/rokiyama/gqlgen-todos
go: creating new go.mod: module github.com/rokiyama/gqlgen-todos

$ go get github.com/99designs/gqlgen
go get: added github.com/99designs/gqlgen v0.13.0

$ go get github.com/vektah/gqlparser/v2
go get: upgraded github.com/vektah/gqlparser/v2 v2.1.0 => v2.2.0

$ go run github.com/99designs/gqlgen init
validation failed: packages.Load: /Users/rokiyama/projects/gqlgen-todos/graph/prelude.resolvers.go:19:44: __DirectiveResolver not exported by package generated

Exec "go run ./server.go" to start GraphQL server

$ go run ./server.go
# github.com/rokiyama/gqlgen-todos/graph
graph/prelude.resolvers.go:19:34: cannot refer to unexported name generated.__DirectiveResolver
```

my go.mod:

```golang
module github.com/rokiyama/gqlgen-todos

go 1.16

require (
        github.com/99designs/gqlgen v0.13.0 // indirect
        github.com/vektah/gqlparser/v2 v2.2.0 // indirect
)
```

**Comment by diffuse** [comment]
Looks like this may be an issue with [gqlparser v2.2.0](https://github.com/vektah/gqlparser/tree/v2.2.0) (released 8 days ago as of this post).  I haven't looked into the specifics, but I just was able to replicate the behavior you mention on my system, then fix it with the resolution mentioned [here](https://stackoverflow.com/questions/67125011/gqlgen-directiveresolver-not-exported-by-package-generated).

Basically, in your `go.mod` file:
- rollback the version of gqlparser from `github.com/vektah/gqlparser/v2 v2.2.0` to `github.com/vektah/gqlparser/v2 v2.1.0`
- delete `graph/prelude.resolvers.go`
- rerun `go run github.com/99designs/gqlgen init`

Credit to [FreezeLook](https://stackoverflow.com/users/15710895/freezelook) for this solution.

**Comment by diffuse** [comment]
Looks like this change is being discussed over in https://github.com/vektah/gqlparser/issues/154.

**Comment by rokiyama** [comment]
@diffuse The solution worked for me. Thank you.

**Comment by wangzuo** [comment]
I guess https://github.com/99designs/gqlgen/blob/master/graphql/introspection/introspection.go#L12 is missing in `v0.13.0`, so a new release will this issue #1534

**Comment by frederikhors** [comment]
@KellyLSB please tell me if this is still an issue with v0.16.0 and I'll re-open it.
